src/background.js

```diff
@@ -28,6 +34,46 @@ browser.browserAction.onClicked.addListener(() => {
     openOverview()
 })
 
+
+browser.bookmarks.onCreated.addListener((id, bookmarkInfo) => {
+    const samePageCandidates = (await findPagesByUrl({url})).rows.map(row => row.doc)
+    if (samePageCandidates.length <= 0) {
+        const favIconBlob = await dataURLToBlob(pageData.favIconURI)
+        const freezeDryBlob = new Blob([pageData.freezeDryHTML], {type: 'text/html;charset=UTF-8'})
+        try {
+            const pageData = await fetchPageData(bookmarkInfo.url, {
+                                                    includeFreezeDry: true,
```

Comment:
Freeze-dry should only be included and attached to the page doc if the `freeze-dry-bookmarks` boolean is set in local storage (handled in Options page). But maybe just leave that to last as it should just be some little conditions. Apart from that, looking good so far!